apps/frontend/src/hooks/__tests__/search.test.ts

```diff
@@ -1,141 +1,187 @@
-import { renderHook } from '@testing-library/react-hooks';
+import { renderHook, act } from '@testing-library/react-hooks';
 import { MemoryRouter } from 'react-router-dom';
+import { searchQueryParam } from '@asap-hub/routing';
+import { waitFor } from '@testing-library/dom';
 
 import { useSearch } from '../search';
 import { usePagination, usePaginationParams } from '../pagination';
 
 describe('useSearch', () => {
```

Comment:
restructured this test file, it had like a hundred test cases flat in one describe block with no discernible order

Reply:
and also added new tests at the bottom ofc